Hearthstone Deck Tracker/Hearthstone/GameV2.cs

```diff
@@ -301,7 +307,9 @@ public void SnapshotBattlegroundsBoardState()
 				.Select(x => x.Clone())
 				.ToArray();
 			Log.Info($"Snapshotting board state for {opponentHero.Card.Name} with {entities.Length} entities");
-			LastKnownBattlegroundsBoardState[opponentHero.CardId] = new BoardSnapshot(entities, GetTurnNumber());
+			LastKnownBattlegroundsBoardState[GetCorrectLastKnownBoardStateCardId(opponentHero.CardId)] = new BoardSnapshot(entities, GetTurnNumber());
 		}
+
+		internal string GetCorrectLastKnownBoardStateCardId(string cardId) => _lastKnownBoardStateLookup.TryGetValue(cardId, out var mapped) ? mapped : cardId;
```

Comment:
`Dictionary.TryGetValue` will throw an `ArgumentNullException` when cardId is null, so you should add a check for that.